Src/library/base/elks/kernel/file.e

```diff
@@ -2093,6 +2126,120 @@ feature {NONE} -- Implementation
 			"eif_file_access_date"
 		end
 
+	eif_temp_file (a_name_template: POINTER): INTEGER
+			-- Create a unique temporary file name and return a file descriptor.
+			--| UX platforms: The `mkstemp()' function generates a unique temporary filename from
+       		--|	template, creates and opens the file, and returns an open file descriptor for the file.
+		note
+			EIS: "name=Creates a unique temporary file", "src=http://man7.org/linux/man-pages/man3/mkstemp.3.html", "protocol=uri"
+		external
+			"C inline use %"eif_file.h%""
+		alias
+			"[
+			#ifdef EIF_WINDOWS
+				#include <fcntl.h>
+				#ifdef _MSC_VER
+				#define open _open
+				#define close _close
+				#define EINVAL 22 /* Invalid argument  */
+				#define EEXIST 17 /* File exists */
+
+				static const char letters[] =
+				"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+				/* Generate a temporary file name based on TMPL.  TMPL must match the
+				   rules for mk[s]temp (i.e. end in "XXXXXX").  The name constructed
+				   does not exist at the time of the call to mkstemp.  TMPL is
+				   overwritten with the result.  */
+				  int len;
+				  char *tmpl;
+				  char *XXXXXX;
+				  tmpl = $a_name_template;
+				  static unsigned long long value;
+				  unsigned long long random_time_bits;
+				  unsigned int count;
+				  int fd = -1;
+				  int save_errno = errno;
+
+				    /* A lower bound on the number of temporary files to attempt to
+				     generate.  The maximum total number of temporary file names that
+				     can exist for a given template is 62**6.  It should never be
+				     necessary to try all these combinations.  Instead if a reasonable
+				     number of names is tried (we define reasonable as 62**3) fail to
+				     give the system administrator the chance to remove the problems.  */
+				  #define ATTEMPTS_MIN (62 * 62 * 62)
+
+				  /* The number of times to attempt to generate a temporary file.  To
+				     conform to POSIX, this must be no smaller than TMP_MAX.  */
+				#if ATTEMPTS_MIN < TMP_MAX
+				  unsigned int attempts = TMP_MAX;
+				#else
+				  unsigned int attempts = ATTEMPTS_MIN;
+				#endif
+
+				  len = strlen (tmpl);
+				  if (len < 6 || strcmp (&tmpl[len - 6], "XXXXXX")){
+				      errno = EINVAL;
+				      return -1;
+				  }
+
+				  /* This is where the Xs start.  */
+				  XXXXXX = &tmpl[len - 6];
+
+				  /* Get some more or less random data.  */
+				  {
+				  SYSTEMTIME    stNow;
+				  FILETIME ftNow;
+
+				  // get system time
+				  GetSystemTime(&stNow);
+				  stNow.wMilliseconds = 500;
+				  if (!SystemTimeToFileTime(&stNow, &ftNow))  {
+				    errno = -1;
+				    return -1;
+				  }
+
+				    random_time_bits = (((unsigned long long)ftNow.dwHighDateTime << 32)
```

Comment:
there are a few wrong indentations.
Please reindent correctly, and there are a few empty lines that could be removed ... to have smaller body.

Reply:
updated.